packages/firebase_snippets_app/lib/snippets/auth.dart

```diff
@@ -0,0 +1,622 @@
+// ignore_for_file: non_constant_identifier_names, avoid_print
+
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:crypto/crypto.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:firebase_snippets_app/snippets/snippet_base.dart';
+import 'package:flutter/foundation.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:sign_in_with_apple/sign_in_with_apple.dart';
+import 'package:twitter_login/twitter_login.dart';
+
+class AuthenticationSnippets implements DocSnippet {
+  @override
+  void runAll() {
+    getStarted_authStateChanges();
+    getStarted_idTokenChanges();
+    getStarted_userChanges();
+    manageUsers_getProfile();
+    passwordAuth_createPasswordBasedAccount();
+    emailLinkAuth_sendAuthLinkEmail();
+    social_google_sign_in();
+    social_twitterAuth();
+    social_facebookAuth();
+    phoneAuth_verifyPhoneNumber();
+    customAuth_withFirebase();
+    anonymousAuth_withFirebase();
+  }
+
+  void getStarted_authStateChanges() async {
+    // [START get_started_auth_state_changes]
+    FirebaseAuth.instance.authStateChanges().listen((User? user) {
+      if (user == null) {
+        print('User is currently signed out!');
+      } else {
+        print('User is signed in!');
+      }
+    });
+    // [END get_started_auth_state_changes]
+  }
+
+  void getStarted_idTokenChanges() async {
+    // [START get_started_id_token_changes]
+    FirebaseAuth.instance.idTokenChanges().listen((User? user) {
+      if (user == null) {
+        print('User is currently signed out!');
+      } else {
+        print('User is signed in!');
+      }
+    });
+    // [END get_started_id_token_changes]
+  }
+
+  void getStarted_userChanges() async {
+    // [START get_started_user_changes]
+    FirebaseAuth.instance.userChanges().listen((User? user) {
+      if (user == null) {
+        print('User is currently signed out!');
+      } else {
+        print('User is signed in!');
+      }
+    });
+    // [END get_started_user_changes]
+  }
+
+  void manageUsers_getProfile() async {
+    // [START manage_users_get_profile]
+    final user = FirebaseAuth.instance.currentUser;
+    if (user != null) {
+      // Name, email address, and profile photo URL
+      final name = user.displayName;
+      final email = user.email;
+      final photoUrl = user.photoURL;
+
+      // Check if user's email is verified
+      final emailVerified = user.emailVerified;
+
+      // The user's ID, unique to the Firebase project. Do NOT use this value to
+      // authenticate with your backend server, if you have one. Use
+      // User.getIdToken() instead.
+      final uid = user.uid;
+    }
+    // [END manage_users_get_profile]
+
+    // [START manage_users_get_provider_specific_profile]
+    if (user != null) {
+      for (final providerProfile in user.providerData) {
```

Comment:
I hardly ever remember that we have a snippet for this. Good to see it in Flutter too tough! 👍